### PR TITLE
Pct share sanity check

### LINKED ISSUE #1806: Verify that the output of each `prod` pipeline is exactly the same as the `dev` pipeline.
## Is your feature request related to a problem? Please describe.

Recently, we had a strange situation where the output of our prod pipeline was not exactly the same as the output of the dev pipeline. This should never happen and if it does we should fail to update the API and investigate. 

## Describe the solution you'd like

We should add a new Airflow job that runs right before the `exporter` step on each `prod` pipeline that ensures that the updated prod big query tables are exactly the same as the dev big query tables.

We can utilize the logic in the https://github.com/SatcherInstitute/health-equity-tracker/blob/main/e2e_tests/scripts/ensure_datasets_equal.py script to do this.

## Additional context

An important thing to note is this is the first time we will have a diversion in the logic between the dev and prod pipelines. In my opinion we should handle this in the step itself. so the actual airflow pipelines will be the same, but the `verify` step, or whatever we choose to call the output of this, will just always pass on the `dev` pipeline.

### PR DESCRIPTION
**Pull request by eriwarr**
## Description

This PR includes a fix to ensure that the sum of percentage share values for a particular demographic, location and time on a given data frame are within the expected range (99-101), as described in issue #1805.

- Installed a new python module to contain the sanity checks. 
- Added the sanity check to the dag directory. 
- Note: Airflow could not import the sanity check module when running the Airflow server, so I included the logic in the Airflow directory. There is likely a way to use the Python Virtual Environment Operator so that the `sanity_check.py` file is not duplicated, but I will revisit it. 

## Motivation and Context

- Fixes #1806

## How has this been tested?

- Ran pytests on `tests/sanity_checks`.
- Ran tests on Airflow server. 

## Screenshots (if appropriate):
Airflow error example
airflow_error

Succesful run
succesful_run

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality or updates content)
- Chore (developer productivity fix that doesn't affect the user)

### CODE REVIEW

File: python/tests/sanity_checks/test_sanity_check.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+test_dtype = {
+    'county_fips': str,
+    'state_fips': str,
+}
+
+
+def testGenerateCountyDatasetAge():
+    df = pd.read_json(CDC_RESTRICTED['age_county'], dtype=test_dtype)
+    result = check_pct_values(df)
+    return result
```
Comment (joshzarrabi) [inline python/tests/sanity_checks/test_sanity_check.py:26]: I think these should be `assert`s. You can do it all in one line: like `assert check_pct_values(df)`

File: python/tests/sanity_checks/test_sanity_check.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+TEST_DIR = os.path.join(THIS_DIR, os.pardir, "data", "sanity_checks")
+
+CDC_RESTRICTED = {
+    'age_county': os.path.join(TEST_DIR, 'cdc_restricted_age_county.json'),
+    'sex_county_time': os.path.join(TEST_DIR, 'cdc_restricted_sex_county_time.json'),
+    'sex_county': os.path.join(TEST_DIR, 'cdc_restricted_sex_county.json'),
+    'sex_national': os.path.join(TEST_DIR, 'cdc_restricted_sex_national.json'),
+    'sex_state': os.path.join(TEST_DIR, 'cdc_restricted_sex_state.json')
+}
+
```
Comment (joshzarrabi) [inline python/tests/sanity_checks/test_sanity_check.py:16]: I think in this file you should also test an error case. Like a case where you expect it to fail and error. See https://github.com/SatcherInstitute/health-equity-tracker/blob/89f6457cf5f1a4dcef520cfd349ce73a0de61025/python/tests/ingestion/test_dataset_utils.py#L278-L281 as an example of how to do that.

File: python/tests/sanity_checks/test_sanity_check.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+def testGenerateCountyDatasetAge():
+    df = pd.read_json(CDC_RESTRICTED['age_county'], dtype=test_dtype)
+    result = check_pct_values(df)
+    return result
+
+
+def testGenerateCountyDatasetSexTime():
+    df = pd.read_json(CDC_RESTRICTED['sex_county_time'], dtype=test_dtype)
```
Comment (joshzarrabi) [inline python/tests/sanity_checks/test_sanity_check.py:30]: Nice job testing time

File: python/sanity_checks/build/lib/sanity_checks/sanity_check.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import pandas as pd
```
Comment (joshzarrabi) [inline python/sanity_checks/build/lib/sanity_checks/sanity_check.py:1]: I dont think this file should be checked in at all.

File: python/sanity_checks/sanity_check.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import pandas as pd
+import ingestion.standardized_columns as std_col
+
+
+def generate_cols(df: pd.DataFrame):
```
Comment (joshzarrabi) [inline python/sanity_checks/sanity_check.py:5]: Do we use this file? It looks like you copied the code to the `airflow/dags/sanity_check.py` file and use that in the pipeline. We should use the code that we run our tests on.

**Review by joshzarrabi — CHANGES_REQUESTED** [review]

**Review by benhammondmusic — APPROVED** [review]
looks good to me; i've started doing screenshots of bigquery / airflow / backend stuff for my backend PRs as I think it can be helpful when reviewing and referencing exactly what these backend changes are doing. Maybe consider taking a few screenshots of the printout from a successful run and a run where things aren't adding up properly

File: airflow/dags/util.py [hunk 6]
```diff
@@ -112,6 +116,22 @@ def service_request(url: str, data: dict, **kwargs):
         raise Exception('Failed response code: {}'.format(err))
 
 
+def sanity_check_request(dataset_id: str):
+    bq_client = bigquery.Client()
+
+    tables = bq_client.list_tables(dataset_id)
+    for table in tables:
+        table_name = "{}.{}.{}".format(
```
Comment (joshzarrabi) [inline airflow/dags/util.py:124]: Would you mind switching this to an `fstring`

File: airflow/dags/test_sanity_check.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import os
```
Comment (joshzarrabi) [inline airflow/dags/test_sanity_check.py:1]: This looks good but I think you should put it in a `test` directory

**Review by joshzarrabi — APPROVED** [review]
Thank you Eric!

**Comment by eriwarr** [comment]
> looks good to me; i've started doing screenshots of bigquery / airflow / backend stuff for my backend PRs as I think it can be helpful when reviewing and referencing exactly what these backend changes are doing. Maybe consider taking a few screenshots of the printout from a successful run and a run where things aren't adding up properly
> looks good to me; i've started doing screenshots of bigquery / airflow / backend stuff for my backend PRs as I think it can be helpful when reviewing and referencing exactly what these backend changes are doing. Maybe consider taking a few screenshots of the printout from a successful run and a run where things aren't adding up properly

Great point! I'll update the PR to include them.

**Pull request merged**